client/modules/i18n/main.js

```diff
@@ -107,7 +106,9 @@ Meteor.startup(() => {
         if (result) {
           const locale = result;
           locale.language = getBrowserLanguage();
-          moment.locale(locale.language);
+          import("moment").then(moment => {
+            moment.locale(locale.language);
+          });
```

Comment:
This is going to have the effect of loading moment right away during the page load, anyway. Won't be in the main bundle, but it still seems like we could get rid of this so that there is less chance of it ever having to load for some people.

All this is doing is setting the locale for moment to use. So that is something we can do right after we load moment at some later point in time.

Therefore, I would add a file somewhere that exports an `importMoment` function like:

```js
export default async function importMoment() {
  const moment = await import("moment");
  moment.locale(locale.language); // get the locale from somewhere where this "shop/getLocale" callback has saved it
  return moment;
}
```

And then change all other `import("moment")` to use our custom `importMoment()`

Reply:
I had a similar initial thought, placing a `getMoment()` in a central location, and ended up with a function inside of `Reaction.`, so we could call `Reaction.getMoment()`, but after chatting to a few people realized that would be called of the bat when `Reaction.` was built, and went this other direction.

Where's a good location for this `importMoment()` function to live? `/lib/api/helpers`

Reply:
I think it would be needed only in client code (`/lib` loads on server too), and also I'm trying to insist that all new code go within `/imports` so that it does not load anywhere unless we import it there. I'm a little unsure about what pattern we want to use, given the current idea of "plugin". @aaronjudd @spencern ? If we have functions we want to have available for import in other files, but not really a plugin or something that needs `registerPackage`, where should those go? Can we make `/imports/helpers`?

If we stick with the pattern of putting all the helpers on the `Reaction` object, then I think we'll end up loading more code than we need to in some cases. That's why I'm thinking we just stick them somewhere in `/imports` and import directly where needed.

Reply:
I agree that moving all code to `/imports` is the direction we should be going. I'm okay with some `helpers` living in their own package. I can't see any reasons that getting the moment lib should be attached to the `Reaction` object right now.

From what I remember, using the Reaction object for helpers like this in the past was because the previous pattern (pre `import`) was that we had several globals, so moving them to the `Reaction` namespace was a way of cleaning that up.